mempool/src/core_mempool/index.rs

```diff
@@ -255,6 +256,100 @@ impl TimelineIndex {
     }
 }
 
+pub struct MultiBucketTimelineIndex {
+    timelines: Vec<TimelineIndex>,
+    bucket_mins: Vec<u64>,
+}
+
+impl MultiBucketTimelineIndex {
+    pub(crate) fn new(bucket_mins: Vec<u64>) -> anyhow::Result<Self> {
+        anyhow::ensure!(!bucket_mins.is_empty(), "Must not be empty");
+        anyhow::ensure!(bucket_mins[0] == 0, "First bucket must start at 0");
```

Comment:
probably also ensure it's sorted or something

Reply:
This is done in the for loop below. Unfortunately `.is_sorted()` is still experimental